File: AccountantWeb/Models/Report.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Accountant.Models
{
    public class Report
    {
        public int ID { get; set; }

        [Required]
        [DataType(DataType.Date)]
        public DateTime Start { get; set; }

        [Required]
        [DataType(DataType.Date)]
        public DateTime End { get; set; }
        public bool Evaluated { get; set; }

        [DataType(DataType.Date)]
        public DateTime DateOfEvaluation { get; set; }
    }
}

File: AccountantAndroid/app/src/main/java/com/kunkliricsi/accountant/local/entities/Report.java
package com.kunkliricsi.accountant.local.entities;

import java.util.Date;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "reports")
public class Report {

    @PrimaryKey
    public int id;

    @ColumnInfo(name = "start")
    public Date Start;

    @ColumnInfo(name = "end")
    public Date End;

    @ColumnInfo(name = "evaluated")
    public boolean Evaluated;

    @ColumnInfo(name = "dateofevaluation")
    public Date DateOfEvaluation;
}
